src/cloud_provider/kubernetes.rs

```diff
@@ -908,6 +921,43 @@ where
         }
     });
 
+    return match result {
+        Ok(_) => match check_workers_status(kubernetes_config.as_ref(), envs.clone()) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e),
+        },
+        Err(Operation { error, .. }) => Err(error),
+        Err(retry::Error::Internal(e)) => Err(SimpleError::new(SimpleErrorKind::Other, Some(e))),
+    };
+}
+
+pub fn check_workers_status<P>(kubernetes_config: P, envs: Vec<(&str, &str)>) -> Result<(), SimpleError>
+where
+    P: AsRef<Path>,
+{
+    let result = retry::retry(Fixed::from_millis(10000).take(360), || {
+        match kubectl_exec_get_node(kubernetes_config.as_ref(), envs.clone()) {
+            Err(e) => OperationResult::Retry(e),
+            Ok(nodes) => {
+                let mut conditions: Vec<KubernetesNodeCondition> = Vec::new();
+                for node in nodes.items.into_iter() {
+                    conditions.extend(node.status.conditions.into_iter());
+                }
+
+                for condition in conditions.iter() {
+                    if condition.condition_type == "Ready" && condition.status != "True" {
+                        info!("There are still not updated nodes. Updating...");
```

Comment:
I would change this, because you may not be in an upgrade strategy. You may just replace node types or have node issues being managed by SCW